remote-api/src/main/java/org/alfresco/rest/api/impl/CategoriesImpl.java

```diff
@@ -106,17 +105,36 @@ public List<Category> createSubcategories(String parentCategoryId, List<Category
                 .collect(Collectors.toList());
     }
 
-    private NodeRef createCategoryNodeRef(NodeRef parentNodeRef, Category c)
+    @Override
+    public Category updateCategoryById(final String id, final Category fixedCategoryModel)
     {
-        if (StringUtils.isEmpty(c.getName())) {
-            throw new InvalidArgumentException(NOT_NULL_OR_EMPTY);
+        verifyAdminAuthority();
+        final NodeRef categoryNodeRef = nodes.validateOrLookupNode(id, null);
+        if (isNotACategory(categoryNodeRef) || isRootCategory(categoryNodeRef))
+        {
+            throw new InvalidArgumentException(NOT_A_VALID_CATEGORY, new String[]{id});
         }
-        return categoryService.createCategory(parentNodeRef, c.getName());
+
+        final Category originalCategory = mapToCategory(categoryNodeRef);
+        compareAndVerifyCategoryFields(fixedCategoryModel, originalCategory);
```

Comment:
As I commented in the `Categories` interface, I believe we should either say that we ignore all fields other than `name` or that we will return error (400) when any fields other than `name` are passed in the request so I think this check should either not be triggered at all or just verify if there are any fields other than `name` (if so - we should call the same check in create category).